t/widgets/render-methods.lisp

```diff
@@ -32,11 +32,52 @@
 (deftest test-widget-rendering-as-a-div
   (let ((widget (make-instance 'simple-widget)))
     (is-html (render widget)
-             "<div class=\"widget simple-widget\"><p>Hello world
-</div>")))
+             "<div class=\"widget simple-widget\"><p>Hello world</div>")))
 
 
 (deftest test-widget-with-custom-tag
   (let ((widget (make-instance 'widget-with-custom-tag)))
     (is-html (render widget)
              "<h1 class=\"widget widget-with-custom-tag\"><p>Hello world</h1>")))
+
+
+(defwidget widget-inside-table ()
+  ())
+
+(defmethod render ((widget widget-inside-table))
+  (with-html
+    (:p "foo bar")))
+
+(deftest test-widget-nested-into-table
+  (let ((widget (make-instance 'widget-inside-table)))
+    (is-html
+     (with-html
+       (:table
+        (:tr
+         (:td
+          (render widget)))))
+     ;; with (*print-pretty* nil) because indentation mixed of spaces and tab
+     ;; is difficult to get correctly.
+     ;; Even with it we need this weird indentation of 3 spaces.
+     "<table><tr><td><div class=\"widget widget-inside-table\"><p>foo bar</div></table>"
+     ;TODO: missing closing tags ??
```

Comment:
HTML allows some tags to not have a closing part for example, for `<p>`. XHTML has more strict rules.